src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,111 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int ABSENT_INDEX = -1;
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int SHIFT = 1;
+    private static final int START = 0;
+    private Object[] elementData = new Object[DEFAULT_CAPACITY];
+    private int size;
+
     @Override
     public void add(T value) {
-
+        if (size < elementData.length) {
+            elementData[size] = value;
+        } else {
+            insert(value, size, expand());
```

Comment:
Do we really need to use arraycopy method to just add one element to the end?
Let's refactor your solution and use arraycopy only once in add by the index

Reply:
Done!